src/dune_rules/utop.ml

```diff
@@ -93,6 +93,57 @@ let libs_and_ppx_under_dir sctx ~db ~dir =
                     , pps )
                 else
                   (acc, pps))
+            | Dune_file.Executables exes -> (
+                let* libs =
+                  let open Memo.Build.O in
+                  let* compile_info =
+                    let project = Scope.project d.scope in
+                    let dune_version = Dune_project.dune_version project in
+                    let+ pps =
+                      Resolve.Build.read_memo_build
+                        (Preprocess.Per_module.with_instrumentation
+                           exes.buildable.preprocess
+                           ~instrumentation_backend:
+                             (Lib.DB.instrumentation_backend (Scope.libs d.scope)))
+                      >>| Preprocess.Per_module.pps in
+                    Lib.DB.resolve_user_written_deps_for_exes
+                      db exes.names exes.buildable.libraries
+                      ~pps ~dune_version
+                      ~allow_overlaps:exes.buildable.allow_overlapping_dependencies
+                      ~forbidden_libraries:exes.forbidden_libraries in
+                  let+ available = Lib.Compile.direct_requires compile_info in
+                  Resolve.peek available in
+                match libs with
+                | Error () -> Memo.Build.return (acc, pps)
+                | Ok libs ->
+                  Memo.Build.List.fold_left libs
+                    ~init:(acc,pps)
+                    ~f:(fun (acc, pps) lib ->
+                      let info = Lib.info lib in
+                      (* Only select libraries that are not implementations.
+                         Implementations are selected using the default implementation
+                         feature. *)
+                      let not_impl = Option.is_none (Lib_info.implements info) in
```

Comment:
There's no need for the `not_impl` check since executables are going to select implementations for all virtual libraries they use.

Reply:
Sure, I see - removed the impl check.